src/runtime/runtime.cs

```diff
@@ -1026,44 +1030,69 @@ internal static bool PyObject_IsIterable(IntPtr pointer)
             return tp_iter != IntPtr.Zero;
         }
 
-        [DllImport(_PythonDll, CallingConvention = CallingConvention.Cdecl)]
-        internal static extern int PyObject_HasAttrString(IntPtr pointer, [MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(Utf8Marshaler))] string name);
+        
+        internal static int PyObject_HasAttrString(BorrowedReference pointer, string name)
+        {
+            using var namePtr = new StrPtr(name, Encoding.UTF8);
+            return Delegates.PyObject_HasAttrString(pointer, namePtr);
+        }
 
-        [DllImport(_PythonDll, CallingConvention = CallingConvention.Cdecl)]
-        internal static extern IntPtr PyObject_GetAttrString(IntPtr pointer, [MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(Utf8Marshaler))] string name);
+        internal static IntPtr PyObject_GetAttrString(IntPtr pointer, string name)
+        {
+            IntPtr nameMem = Marshal.StringToHGlobalAnsi(name);
+            try
+            {
+                return Delegates.PyObject_GetAttrString(pointer, nameMem);
+            }
+            finally
+            {
+                Marshal.FreeHGlobal(nameMem);
+            }
+        }
 
-        [DllImport(_PythonDll, CallingConvention = CallingConvention.Cdecl)]
-        internal static extern int PyObject_SetAttrString(IntPtr pointer, [MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(Utf8Marshaler))] string name, IntPtr value);
+        
+        internal static IntPtr PyObject_GetAttrString(IntPtr pointer, IntPtr name) => Delegates.PyObject_GetAttrString(pointer, name);
```

Comment:
Is this used somewhere? If not, I'd argue we should remove it from here, the string handling is not trivial.

Reply:
I changed the signature to accept `StrPtr` instead